salt/states/ssh_auth.py

```diff
@@ -286,6 +291,16 @@ def present(
             if len(comps) == 3:
                 comment = comps[2]
 
+    if 'ssh_keys' in kwargs:
```

Comment:
Can we expose this so that it can be used directly as well, instead of only through mod_aggregate?

Reply:
Ok, removing the ssh keys is not the purpose of mod_aggregate.

The purpose of mod_aggregate is to combine all of the calls into just one call.

See how it is used in pkg.installed where it makes only one call to the system package manager, instead of once per package state.

I am not sure that we should make this change if it is going to start removing keys that are not managed by the aggregate.  @saltstack/team-core opinions?

Reply:
Agreed. Hence the initial "I don't know if this is kosher" comment. 

In order to perform authorized_keys management where keys not managed by salt are removed, we would need to know about all the keys that salt is managing. Today that means a series of `ssh_auth.present` stanzas. Because they are indepent, it forces the use of mod_aggregate. This overloads the use of mod_aggregate. In the pkg state it aggregates calls while in this use it aggregates and allows the multiple calls to more completely manage the authorized_keys file. Possibly a hack, or possibly awesome?!

If there are cleaner ways of accomplishing this, I'm for that as well. But a definitely a review is needed before any potential merge.

Reply:
yeah, i just want to have more input from the rest of the core team to see if we are ok with this.